tools/niminst/buildsh.tmpl

```diff
@@ -123,7 +123,15 @@ case $ucpu in
   *power*|*ppc* )
     mycpu="powerpc" ;;
   *mips* )
-    mycpu="mips" ;;
+    mycpu="$("$CC" -dumpmachine | sed 's/-.*//')"
```

Comment:
`uname -m` contains the processor architecture and not the architecture the C compiler targets. This is important on mips because a lot of mips machines run 64-bit kernels but have 32-bit userspace. Using `uname -m` will therefore give the wrong result.

Side note: this is one of the reasons things like autotools have insanely complicated scripts for determining the target architecture.